Algorithms/Search/C++/Binary search.cpp

```diff
@@ -1,36 +1,23 @@
 #include <iostream>
 using namespace std;
-
-
-int binarySearch(int arr[], int l, int r, int x)
+int binarySearch(int arr[], int l, int r, int x) //Binary Search function to find element x in array arr[]
 {
     if (r >= l) {
-        int mid = l + (r - l) / 2;
-  
-
-        if (arr[mid] == x)
-            return mid;
-
-
-        if (arr[mid] > x)
-            return binarySearch(arr, l, mid - 1, x);
-
-
-        return binarySearch(arr, mid + 1, r, x);
+        int mid = l + (r - l) / 2; //selection of mid element
+        return (arr[mid]==x?mid:arr[mid]>x?binarySearch(arr, l, mid - 1, x):binarySearch(arr, mid + 1, r, x));
```

Comment:
Hi @Davidgokimmung 
If you want to use a ternary operator, try adding spaces around operators `==`, `?` etc to make it a bit more readable and maybe split it into two lines.